app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt

```diff
@@ -50,12 +53,70 @@ class DrawableAdapter(private val context: Context?, private val list: List<Draw
         }
 
         fun bind(drawableInfo: DrawableInfo) {
-            image.setImageDrawable(drawableInfo.image)
+            image.setImageBitmap(trim((drawableInfo.image as BitmapDrawable).bitmap))
 
             card.background = when {
                 selectedPosition != -1 && selectedPosition == adapterPosition -> context?.resources?.getDrawable(R.color.colorAccent)
                 else -> context?.resources?.getDrawable(android.R.color.transparent)
             }
+            image.setColorFilter((when {
+                selectedPosition != -1 && selectedPosition == adapterPosition -> context?.resources?.getColor(android.R.color.white)
+                else -> context?.resources?.getColor(android.R.color.black)
+            }) ?: Color.parseColor("#000000"))
+        }
+
+        fun trim(source: Bitmap): Bitmap {
```

Comment:
Should be no need of any algorithm, just reduce the padding on the button. If it is 0, apply negative margin

Reply:
@iamareebjamal ok sir, let me try this as soon as I reach home

Reply:
@iamareebjamal, Btw sir there is no padding given to the button. The drawables are in 40x11. The larger width was the issue which caused the image to fit inside the imageview with the transparent pixels. These transparent pixels made the image look smaller

Reply:
@iamareebjamal I will still have to calculate the amount of margin to reduce for every bitmap as the transparent pixels vary from image to image

Reply:
@iamareebjamal sir, If you look in the current screenshot. Check the clipart at (2,1). It is much wider than the rest. All of them were the same with transparent pixels on all sides, which looked uneven. So I just trimmed those transparent pixels and then set the bitmap.

Reply:
@iamareebjamal good morning sir, as the padding was 0, I tried applying negative margin of -20dp, It did zoom in but the images which were wider had its end chopped off. So if we plan to give negative margin, it needs to be dynamic according to the transparent pixels for which we need to create an algorithm anyways. After giving the negative margin, the resulting image will be too smooth and hence, we still need to perform a scale which I just added to this function. Please let me know your thoughts on this.
Regards

Reply:
@iamareebjamal please review